install/kubernetes/helm/istio/charts/pilot/templates/gateway.yaml

```diff
@@ -74,3 +74,72 @@ spec:
     hosts:
     - "*"
 {{- end }}
+---
+{{- if .Values.global.multiClusterUsingGateway }}
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-multicluster-gateway
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+    istio: mcgateway
+  servers:
+  - hosts:
+    # This is a dummy port to keep AWS LB happy.
+    # DO NOT REMOVE.
+    - "*.local"
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+  - hosts:
+    - "*.local"
+    port:
+      name: tls
+      number: 443
+      protocol: TLS
+    tls: {}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: istio-multicluster-virtualservice
+  namespace: {{ .ReleaseNamespace }}
+spec:
+  gateways:
+  - istio-multicluster-gateway
+  hosts:
+  - "*.local"
+  # We need a TLS block to setup the filter chain
+  # The actual routing happens using the SNI filter
+  # Do not add any route for port 80. It will default to blackhole cluster.
+  tls:
+  - match:
+    - port: 443
+      sniHosts:
+      - "*.local"
+    route:
+    - destination:        
+        host: non.existent.cluster
```

Comment:
I don't think I understand this destination... what is the intended destination for this SNI?  would this be resolved to black hole cluster due to not routable route?

Reply:
Yep. This is just to get Pilot to generate a tcp proxy filter chain for this port. The actual routing happens in a different filter.